File: src/javascript/compress_ew.js
const assert = require("assert");

//     v 74 chars, no regex /\/\/\:
compress=s=>[...s].reduce(([r,c],v,i,a)=>(v==a[i+1]?[r,c+1]:[r+c+v,1]),['',1])[0]

assert.equal(compress("AAABBAAC"), "3A2B2A1C");

File: src/python/compress_ew.py
def compress(s='', r="", c=1):
    for p, v in zip(s, s[1:] + '0'):
        # print(c, p, v, r)
        if p == v:
            c += 1
        else:
            r += str(c) + p
            c = 1

    return r


assert compress("AAABBAAC") == "3A2B2A1C"
